core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -1523,23 +1534,21 @@
                             [{'application_time_start `(~'cast-tstz ~(cond
                                                                        (= :all-application-time app-time-extents) app-start-sym
                                                                        app-from-expr `(~'greatest ~app-start-sym ~app-from-expr)
-                                                                       app-time-as-of-now? `(~'greatest ~app-start-sym (~'current-timestamp))
+                                                                       (not all-app-time?) `(~'greatest ~app-start-sym (~'current-timestamp))
```

Comment:
@jarohen If you have an idea how to get rid of this `greatest` here, we can get rid of `all-app-time?` in the `plan/*opts*`.